### PR TITLE
User interface initial design

### PR DESCRIPTION
**Pull request by ak-14**

### CODE REVIEW

File: app/assets/stylesheets/modal_images.scss [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+#img {
+  border-radius: 5px;
+  cursor: pointer;
+  transition: 0.3s;
+}
+
+#img:hover {opacity: 0.7;}
+
+/* The Modal (background) */
+.modal {
```
Comment (iFahad7) [inline app/assets/stylesheets/modal_images.scss]: Remove modal functionality for now

File: app/assets/stylesheets/modal_images.scss [hunk 2]
```diff
@@ -0,0 +1,87 @@
+/* Style the Image Used to Trigger the Modal */
```
Comment (iFahad7) [inline app/assets/stylesheets/modal_images.scss]: We don't need this for now, delete it.

File: app/assets/stylesheets/main.scss [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+  display: inline-block;
+  opacity: 0.65;
+  border-radius: 5px;
+  &:hover {
+    background-color: black;
+    color: white;
+  }
+}
+
+// Search page
```
Comment (iFahad7) [inline app/assets/stylesheets/main.scss]: Put the '/search' page CSS in a separate file. Generally, each page has its own CSS file.

File: app/controllers/main_controller.rb [hunk 4]
```diff
@@ -0,0 +1,29 @@
+class MainController < ApplicationController
+
+  def index
+  end
+
+  def search
+    if params[:commit].blank?
+      @images = Image.page(params[:page]).per(5)
+    else
+      # @images = Image.joins(:camera).where cameras: { model: params[:camera] }
```
Comment (iFahad7) [inline app/controllers/main_controller.rb]: If we don't need for now, delete it.

File: app/views/main/_images.html.erb [hunk 5]
```diff
@@ -0,0 +1,38 @@
+<div class="images-list">
+  <% @images.each do |image| %>
+    <div class="image">
+      <%= image_tag(image[:thumbnail_url], id: 'img', alt: image[:image_url],
```
Comment (iFahad7) [inline app/views/main/_images.html.erb]: This should be just `<%= image_tag(image[:thumbnail_url], class: 'some-class') %>`

File: app/views/main/_images.html.erb [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+                    onclick: 'show(this)') %>
+      <p>By: <%= link_to image.creator[:name], image[:page_url] %></p>
+    </div>
+  <% end %>
+</div>
+<%= paginate @images unless @images.empty? %>
+
+<!-- The Modal -->
```
Comment (iFahad7) [inline app/views/main/_images.html.erb]: Delete the rest of this file and but the above div in search template

File: app/views/main/index.html.erb [hunk 7]
```diff
@@ -0,0 +1,11 @@
+<div class="cover">
+<div class="main">
```
Comment (iFahad7) [inline app/views/main/index.html.erb]: Indent here

File: app/views/main/search.html.erb [hunk 8]
```diff
@@ -0,0 +1,24 @@
+<div class="search-page">
+    <%= form_tag('/search', method: 'get', id: 'filters', class: 'filters-list') do %>
```
Comment (iFahad7) [inline app/views/main/search.html.erb]: erb files should use 2 space indentation, just like line 23

**Comment by ak-14** [comment]
Done! thanks for the helpful comments bro!

File: app/controllers/main_controller.rb [hunk 9]
```diff
@@ -1,13 +1,14 @@
 class MainController < ApplicationController
 
+  PER_PAGE = 5
+
   def index
   end
 
   def search
     if params[:commit].blank?
       @images = Image.page(params[:page]).per(5)
```
Comment (iFahad7) [inline app/controllers/main_controller.rb]: Forgot to use the `PER_PAGE` constant

**Pull request merged**